packages/realm/src/Object.ts

```diff
@@ -382,9 +382,9 @@ export class RealmObject<T = DefaultObject, RequiredProperties extends keyof Omi
    * and [flatted](https://www.npmjs.com/package/flatted) to stringify Realm entities that have circular structures.
    * @returns A plain object.
    */
-  toJSON(_?: string, cache?: unknown): OmittedRealmTypes<T>;
+  toJSON(_?: string, cache?: unknown): OmittedRealmTypes<T | Unmanaged<T>>;
   /** @internal */
-  toJSON(_?: string, cache = new JSONCacheMap<T>()): OmittedRealmTypes<T> {
+  toJSON(_?: string, cache = new JSONCacheMap<T>()): OmittedRealmTypes<T | Unmanaged<T>> {
```

Comment:
Thank you for your commit.

`Unmanaged<T>` will remap all properties to optional.

If a property is not optional in its `Realm.Object` schema, instances created by that Object will certainly have that property and it will not be optional.

Will this cause the object returned by `toJSON()` to need to be remapped with type definitions in order to be used correctly?

Do you believe OmittedRealmTypes<T> is sufficient?